group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2415,6 +2415,20 @@ private CoordinatorResult<Void, Record> classicGroupJoinExistingMember(
         return EMPTY_RESULT;
     }
 
+    /**
+     * An overload of {@link GroupMetadataManager#completeClassicGroupJoin(ClassicGroup)} used as
+     * timeout operation. It additionally looks up the group by the id and checks the group type.
+     * completeClassicGroupJoin will only be called if the group is CLASSIC.
+     */
+    private CoordinatorResult<Void, Record> completeClassicGroupJoin(String groupId) {
+        if (containsClassicGroup(groupId)) {
+            return completeClassicGroupJoin(getOrMaybeCreateClassicGroup(groupId, false));
```

Comment:
I am not a fan of this pattern because you effectively have to look up the group twice. One option would be to use a try..catch to catch the exception thrown by getOrMaybeCreateClassicGroup. Another option would be to 1) do the lookup, 2) verify non-null and group type and return if it fails.

Reply:
Yes I agree. Let me use the try catch